rs/xnet/payload_builder/src/lib.rs

```diff
@@ -1314,14 +1315,30 @@ impl PoolRefillTask {
                     Ok(slice) => {
                         let res = if witness_begin != msg_begin {
                             // Pulled a stream suffix, append to pooled slice.
-                            pool.lock()
+                            rt_handle
+                                .spawn_blocking(move || {
+                                    pool.lock().unwrap().append(
```

Comment:
Just noticed now that `pool` (`PoolRefillTask::pool`) is an `Arc` around an `std::sync::Mutex`. Isn't there some async-friendly mutex we could be using, instead of blocking runtime threads?

That might even make `spawn_blocking()` unnecessary, as `CertifiedSlicePool::append()` and `put()` are rather lightweight (they merely unpack the protobuf / CBOR and do some basic validation, no verifying certifications).